### PR TITLE
Add CHANGELOG.md entry checker to GHA workflows

### PR DESCRIPTION
**Pull request by therealprof**
r? @adamgreig

Signed-off-by: Daniel Egger <daniel@eggers-club.de>

### CODE REVIEW

File: .github/workflows/changelog.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+on:
+  pull_request_target:
```
Comment (adamgreig) [inline .github/workflows/changelog.yml:2]: Does it need to be `pull_request_target`? On the example for the action they seem to just use a normal `pull_request`.

Comment (therealprof) [reply .github/workflows/changelog.yml:2]: Not sure, it's a verbatim copy from the `stm32f4xx-hal`. 🤷🏻‍♂️

Comment (adamgreig) [reply .github/workflows/changelog.yml:2]: Could you try with just pull_request? If that works it's somewhat more straightforward and safer.

Comment (therealprof) [inline .github/workflows/changelog.yml:2]:
```suggestion
  pull_request:
```

**Comment by adamgreig** [comment]
Thanks, looks like it works. Currently it won't stop bors at all (if I bors+ this now, it should merge, I think) because it doesn't run on bors' push to staging and also bors isn't configured to care about it. I think this sort of makes sense - it should alert reviewers/users that the changelog is missing, but doesn't stop a change being r+'d if the reviewer thinks it's fine. I don't think the "apply label to allow it to pass without changelog" would work under bors anyway, so this is probably the best config. What do you think? Happy to merge as-is.

**Review by adamgreig — DISMISSED** [review]

**Comment by adamgreig** [comment]
Oh, sorry, I didn't notice the force push. What did you change?

**Review by adamgreig — APPROVED** [review]
bors merge

**Comment by therealprof** [comment]
As requested I changed `pull_request_target` to `pull_request`.

**Comment by therealprof** [comment]
> I don't think the "apply label to allow it to pass without changelog" would work under bors anyway, so this is probably the best config. What do you think? Happy to merge as-is.

I agree. Alerting the reviewer to a missing CHANGELOG entry should be okay.

**Pull request merged**